cpp/src/replace/clamp.cu

```diff
@@ -90,18 +90,16 @@ std::unique_ptr<cudf::column> clamp_string_column(strings_column_view const& inp
   // build offset column
   auto offsets_transformer = [lo_itr, hi_itr, lo_replace_itr, hi_replace_itr] __device__(
                                string_view element, bool is_valid = true) {
-    const auto d_lo         = (*lo_itr).first;
-    const auto d_hi         = (*hi_itr).first;
-    const auto d_lo_replace = (*lo_replace_itr).first;
-    const auto d_hi_replace = (*hi_replace_itr).first;
-    const auto lo_valid     = (*lo_itr).second;
-    const auto hi_valid     = (*hi_itr).second;
+    const auto d_lo         = lo_itr->value_or(element);
+    const auto d_hi         = hi_itr->value_or(element);
+    const auto d_lo_replace = *(*lo_replace_itr);
```

Comment:
this can aalso be `->value` right?

Reply:
`optional::value()` has error checking. The dereference operator is the way to access the value w/o error checking.

Reply:
Then the previous line should be reverted?

Reply:
`optional::value_or()` doesn't throw exceptions as it always can return a value ( unlike `value()` ).

Reply:
Just curious. What does `*lo_replace_itr` and `*(*lo_replace_itr)` return if there is no value?

Reply:
The return value is undefined when you have no value. 

No default initialization occurs in these cases so I would expect you will get some random value